### DISCUSSION TITLE
proper matc usage?

### DISCUSSION THREAD

**Discussion by davidwarford** [google/filament#3353] [Q&A] [answered]
I have been trying to compile this material 
```
material {
    name : nonlit,
    requires : [
        color
    ],
    shadingModel : unlit,
    culling : none
}

fragment {
    void material(inout MaterialInputs material) {
        prepareMaterial(material);
        material.baseColor = getColor();
    }
}
```
and for some reason I get no errors and no output

my usage is `C:\Users\david>C:\Users\david\Desktop\bin\matc.exe -o unlit.filamat C:\Users\david\Desktop\bin\unlit.mat`
this is on windows using the current release version of filament.
Is my usage wrong? Is there an error in the material?

**Comment by romainguy** [comment, marked as answer]
I tried your file, it works just fine and I get a proper `.filamat` file. @bejado Given how `matc` is invoked above, maybe we output the file in an unexpected directory?

@davidwarford Have you checked in `C:\Users\david\Desktop\bin\` by any chance? Maybe the file was generated there. I would expect the file to be in `C:\Users\david` though.

**Reply by davidwarford** [reply]
I have checked both directories and it is not in either. I also checked my C drive to see if it was in the base directory and i also tried the same command but with the output going to an absolute path.
im going to try restarting my computer and disabled my antivirus (it seemed to not like matc)

**Reply by romainguy** [reply]
I'd be surprised if `matc` didn't work on Windows at all since we use it as part of the build to generate a bunch of materials that are part of Filament itself.

**Reply by davidwarford** [reply]
i would be too, although that's what it looks like right now. I didn't make this an issue because im sure im either doing something wrong or something on my system is conflicting.

**Reply by davidwarford** [reply]
any ideas for how to diagnose this issue?

**Reply by davidwarford** [reply]
Or actually, in the meantime. is there a way to get unlit materials without providing them as an asset?

**Reply by davidwarford** [reply]
I tried it in WSL and it still does not give output so I got no clue
```
nix@CA1_072216-PC:~$ /mnt/c/Users/david/Desktop/bin/matc -o ~/unlit.filamat /mnt/c/Users/david/Desktop/bin/unlit.mat
nix@CA1_072216-PC:~$ ls
Desktop  Documents  Downloads  Music  Pictures  Public  Templates  Videos  emsdk  emsdk.zip  filament
```

**Reply by bejado** [reply]
I can try running matc on a Windows machine myself later today, but like Romain said, I'd be very surprised if matc didn't work on Windows. Can you try running the command prompt as administrator? Or try setting the output location to a drive other than your C drive?

**Reply by davidwarford** [reply]
i figured out the issue, i didn't save the material in my editor. im going to make a feature request now asking that an error message be added if the input file is empty (or something like that. sorry for the bother and thanks for helping
